tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/BundlePublishFullAdd.java

```diff
@@ -126,13 +129,17 @@ public void execute( IProgressMonitor monitor, IAdaptable info ) throws CoreExce
             {
                 this.portalServerBehavior.setModulePublishState2(
                     new IModule[] { module }, IServer.PUBLISH_STATE_NONE );
+
+                project.deleteMarkers( LiferayServerCore.BUNDLE_COMPILE_MARKER_TYPE, false, 0 );
```

Comment:
There are other situations where we should deleteMarkers, like if the bundle is removed we should remove the markers.